### PR TITLE
book: update refs to release specific books

### PR DESCRIPTION
**Pull request by Prajyot-Parab**
**What this PR does / why we need it**:
- book: update refs to release specific books

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

/area provider/ibmcloud

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note
book: update refs to release specific books
```

### CODE REVIEW

File: docs/book/src/introduction.md [hunk 1]
```diff
@@ -16,6 +16,18 @@ hybrid deployments of Kubernetes.  It is built atop the lessons learned from
 previous cluster managers such as [kops](https://github.com/kubernetes/kops) and
 [kubicorn](http://kubicorn.io/).
 
+<aside class="note">
+
+<h1>Cluster API Provider IBM Cloud documentation versions</h1>
+
+This book documents Cluster API Provider IBM Cloud v0.5. For other versions please see the corresponding documentation:
```
Comment (mkumatag) [inline docs/book/src/introduction.md:23]: are you sure the [cluster-api-ibmcloud.sigs.k8s.io](https://cluster-api-ibmcloud.sigs.k8s.io/) points to the v0.5 release?

Comment (mkumatag) [reply docs/book/src/introduction.md:23]: my understanding is that it points to the main branch which is v0.6 version!

**Review by mkumatag — CHANGES_REQUESTED** [review]

File: docs/book/src/introduction.md (hunk 1 above)
Comment (Prajyot-Parab) [reply docs/book/src/introduction.md:23]: PR raised - https://github.com/kubernetes/k8s.io/pull/5545 to point default url to latest stable version of doc (v0.5)

**Comment by Prajyot-Parab** [comment]
@mkumatag This can be merged now

**Review by mkumatag — APPROVED** [review]
/lgtm

**Pull request merged**